src/main/cpp/profiler.h

```diff
@@ -79,12 +90,18 @@ class Profiler {
 
     Processor *processor;
 
-    SignalHandler handler_;
+    SignalHandler* handler_;
+
+    bool reloadConfig;
```

Comment:
I think option 2 is a better choice here. Are you happy to implement this improve to your patch? Thanks a lot for the contribution btw, it could prove very useful.